### PR TITLE
refactor: Migrate SliceAdder to typescript

### PR DESCRIPTION
**Pull request by EnxDev**
### SUMMARY

Migrate SliceAdder to typescript

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N.A

### TESTING INSTRUCTIONS

- All tests should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/SliceAdder.test.tsx [hunk 1]
```diff
@@ -16,37 +16,48 @@
...
+    lastUpdated: 0,
   };
   const errorProps = {
     ...props,
     errorMessage: 'this is error',
   };
-
   describe('SliceAdder.sortByComparator', () => {
     it('should sort by timestamp descending', () => {
+      console.log(Object.values(props.slices));
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/SliceAdder.test.tsx:60]: can you remove console.log?

Comment (EnxDev) [reply superset-frontend/src/dashboard/components/SliceAdder.test.tsx:60]: > can you remove console.log?

Yes, sorry 😊

File: superset-frontend/src/dashboard/components/SliceAdder.tsx [hunk 2]
```diff
@@ -41,31 +41,40 @@ import {
...
-  userId: PropTypes.number.isRequired,
-  selectedSliceIds: PropTypes.arrayOf(PropTypes.number),
-  editMode: PropTypes.bool,
-  dashboardId: PropTypes.number,
+export type SliceAdderProps = {
+  fetchSlices: (
+    userId?: number,
+    filter_value?: string,
+    sortColumn?: string,
+  ) => void;
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/SliceAdder.tsx:58]: I think it shouldn't return void? Probably some Promise?

**Review by geido — APPROVED** [review]

**Pull request merged**